src/modules/pmc/src/lib/PmcBase.cpp

```diff
@@ -18,20 +18,26 @@ static const std::string g_reportedObjectName = "state";
 static const std::string g_desiredObjectName = "desiredState";
 static const std::string g_packages = "packages";
 static const std::string g_sources = "sources";
+static const std::string g_gpgKeys = "gpgKeys";
 static const std::string g_executionState = "executionState";
 static const std::string g_executionSubState = "executionSubState";
 static const std::string g_executionSubStateDetails = "executionSubStateDetails";
 static const std::string g_packagesFingerprint = "packagesFingerprint";
 static const std::string g_sourcesFingerprint = "sourcesFingerprint";
 static const std::string g_sourcesFilenames = "sourcesFilenames";
-static const std::string g_requiredTools[] = {"apt-get", "apt-cache", "dpkg-query"};
 
-constexpr const char* g_commandCheckToolPresence = "command -v $value";
 constexpr const char* g_commandAptUpdate = "apt-get update";
 constexpr const char* g_commandExecuteUpdate = "apt-get install $value -y --allow-downgrades --auto-remove";
 constexpr const char* g_commandGetInstalledPackageVersion = "apt-cache policy $value | grep Installed";
+constexpr const char* g_commandDownloadGpgKey = "bash -c 'set -o pipefail && curl $url | gpg --dearmor | tee $destination'";
```

Comment:
would `sh -c` be better here? Not all distros have bash.

Reply:
I refactored the command to avoid the bash dependency while at the same time making sure that the piped commands fail when the GPG key cannot be downloaded or is not in the expected format.